### PR TITLE
Rubocop chefspec foodcritic

### PR DESCRIPTION
**Pull request by TD-4242**
I want to open this pull early request as it has a lot of small format changes to satisfy some linting tests.  This should also allow travisci to start doing builds and tests.

I'm having issues getting kitchen to converge everything, the MonitWrapper requires 'waituntil' and I haven't seen where that comes from.  I'll be digging more into that today.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 language: ruby
+bundler_args: --without integration
+rvm:
+  - 2.1.4
```
Comment (mbautin) [inline .travis.yml]: Should this be in the same `rvm:` section as Ruby 2.0.0 and Ruby 2.1.5 below?

File: recipes/spark-install.rb [hunk 2]
```diff
@@ -57,17 +57,54 @@
...
+  end
+elsif node['apache_spark']['local_dir']
+  local_dirs = Array(node['apache_spark']['local_dir'])
+  local_dirs.each do |dir|
+    # Discourage using comma-separated strings in Chef attributes
+    fail "Spark local directory names cannot include a comma: #{dir}" if dir.include?(',')
+  end
+else
+  local_dirs = ['/var/local/spark']
+end
```
Comment (mbautin) [inline recipes/spark-install.rb]: This code was removed in an earlier commit. I am not sure how exactly it could come back in your pull request.

**Comment by mbautin** [comment]
Thank you for your contribution! I will look into integrating this soon. I will also look into test-kitchen issues.

**Comment by mbautin** [comment]
The monit_wrapper waitutil issue should be solved in the most recent version of the monit_wrapper cookbook. Please make sure to delete Berksfile.lock before you re-run `kitchen converge`.

**Comment by mbautin** [comment]
Merging this now anyway -- I'll address the comments I've added above in follow-up commits.

File: recipes/spark-install.rb (hunk 2 above)
Comment (TD-4242) [inline recipes/spark-install.rb]: interesting, probably a mixup when I did a merge this morning.

File: .travis.yml (hunk 1 above)
Comment (TD-4242) [inline .travis.yml]: this looks like a merge thing as well.  let me re-validate and update the PR.

**Comment by TD-4242** [comment]
thanks for merging.  I'm working on getting chefspec stuff up.  Having to mock some things I've never done before with the mixin::shell_out call.  Everything should run including the kitchen test by using either chef exec rake or bundle exec rake, depending on your environment.

**Pull request merged**